### PR TITLE
feat(extension): lw-9532 extend stake pool details drawer with more metadata

### PR DESCRIPTION
**Pull request by vetalcore**
# Checklist

- [x] JIRA -[ LW-9532](https://input-output.atlassian.net/browse/LW-9532)
- [x] Proper tests implemented
- [x] Screenshots added.

# TODO:
- [x] pledge is coming as lovelaces value (design show percentages)
- [x] how to calculate costs per epoch

---

## Proposed solution

add blocks, costs/p epoch, pledge, margin into the stake pool details drawer

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Attach screenshots here if implementation involves some UI changes

Screenshot 2024-01-15 at 14 00 42
Screenshot 2024-01-15 at 14 00 55

### CODE REVIEW

File: packages/cardano/src/ui/components/StakePoolMetricsBrowser/StakePoolMetricsBrowser.tsx [hunk 1]
```diff
@@ -10,54 +9,26 @@ const formatNumericValue = (val: number | string, suffix: number | string): Reac
...
-      <div className={styles.statBody} data-testid="apy-value">
-        {formatNumericValue(apy, '%')}
-      </div>
-    </div>
+    {data.map(({ value, unit, t, testId }) => (
+      <div key={testId} className={styles.statItem} data-testid={testId}>
+        <div className={styles.statHeader} data-testid={`${testId}-title`}>
+          {t}
+        </div>
+        <div className={styles.statBody} data-testid="active-stake-value">
```
Comment (mchappell) [inline packages/cardano/src/ui/components/StakePoolMetricsBrowser/StakePoolMetricsBrowser.tsx:28]:
```suggestion
        <div className={styles.statBody} data-testid={`${testId}-value`}>
```

File: packages/staking/src/features/store/delegationPortfolioStore/mapStakePoolToDisplayData.ts [hunk 2]
```diff
@@ -16,11 +16,13 @@ export const mapStakePoolToDisplayData = ({
 
   return {
     apy: stakePool.metrics?.apy && formatPercentages(stakePool.metrics.apy),
+    blocks: stakePool.metrics?.blocksCreated || '-',
     contact: {
       primary: stakePool.metadata?.homepage,
       ...stakePool.metadata?.ext?.pool.contact,
     },
     cost: `${margin || '-'}% + ${fee}${cardanoCoinSymbol}`,
+    costsPerEpoch: 'costsPerEpoch',
```
Comment (mchappell) [inline packages/staking/src/features/store/delegationPortfolioStore/mapStakePoolToDisplayData.ts:25]: reminder to change for a transformable value

File: apps/browser-extension-wallet/src/features/delegation/stores/createDelegationStore.ts [hunk 3]
```diff
@@ -45,7 +46,10 @@ DelegationStore): stakePoolDetailsSelectorProps => {
       contact: {
         primary: homepage,
         ...ext?.pool.contact
-      }
+      },
+      blocks: blocksCreated || '-',
+      costsPerEpoch: 'costsPerEpoch',
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/features/delegation/stores/createDelegationStore.ts:51]: reminder to change for a transformable value

**Review by mchappell — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/features/delegation/stores/createDelegationStore.ts [hunk 4]
```diff
@@ -45,7 +46,10 @@ DelegationStore): stakePoolDetailsSelectorProps => {
...
+      pledge: Wallet.util.lovelacesToAdaString(pledge.toString())
```
Comment (mchappell) [inline apps/browser-extension-wallet/src/features/delegation/stores/createDelegationStore.ts:52]: 👍

**Review by mchappell — APPROVED** [review]
Nice job @vetalcore!

**Review by wklos-iohk — APPROVED** [review]

**Pull request merged**